core/src/main/scala/org/apache/spark/h2o/H2OContext.scala

```diff
@@ -457,12 +457,25 @@ object H2OContext extends Logging {
    * @return H2O Context
    */
   def getOrCreate(conf: H2OConf): H2OContext = synchronized {
-    val isExternalBackend = conf.runsInExternalClusterMode
-    if (isExternalBackend && !H2OClientUtils.isH2OClientBased(conf)) {
+    if (H2OClientUtils.isH2OClientBased(conf)) {
+      if (instantiatedContext.get() == null)
```

Comment:
qq: Where has the context been created?

Reply:
The context is being created in case of client here:

```
      if (instantiatedContext.get() == null)
        if (H2O.API_PORT == 0) { // api port different than 0 means that client is already running
          val checkedConf = checkAndUpdateConf(conf)
          instantiatedContext.set(new H2OContext(checkedConf))
``` 
The last line

Reply:
oh, ok thanks! I'm blind.